app/Services/GameService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Services;
+
+class GameService {
+    public function __construct( private GameService $gameService)
+    {
+    }
+
+    function getGamesById (?int $id = null)
+    {
+        $games = $this->gameService->getGames();
+
+        foreach ($games as $game) {
+            if ($game->id === $id) {
+                return $game;
+            }
+            return view('games', ['games'=>$this->gameService->getGamesById($id)]);
```

Comment:
this method should return a single game, not a view. the controller calling this method will handle be responsible for the view.